Native application fails with NoSuchMethodException when RestEasy Reactive is used

After migrating to RestEasy Reactive (git revert 428f579b1e86abe4181240d200504d850740844e) the following error is present in the application logs when http://localhost:8080/client/platforms is hit: 

```java
java.lang.RuntimeException: java.lang.NoSuchMethodException: io.quarkus.registry.app.DatabaseRegistryClient.resolveCurrentPlatformsCatalog(java.lang.String)
	at org.jboss.resteasy.reactive.server.spi.ResteasyReactiveResourceInfo.getMethod(ResteasyReactiveResourceInfo.java:65)
	at org.jboss.resteasy.reactive.server.spi.ResteasyReactiveResourceInfo.getGenericReturnType(ResteasyReactiveResourceInfo.java:89)
	at org.jboss.resteasy.reactive.server.core.ResteasyReactiveRequestContext.getGenericReturnType(ResteasyReactiveRequestContext.java:588)
	at org.jboss.resteasy.reactive.server.core.ServerSerialisers.invokeWriter(ServerSerialisers.java:213)
	at org.jboss.resteasy.reactive.server.core.ServerSerialisers.invokeWriter(ServerSerialisers.java:178)
	at org.jboss.resteasy.reactive.server.core.serialization.FixedEntityWriterArray.write(FixedEntityWriterArray.java:28)
	at org.jboss.resteasy.reactive.server.handlers.ResponseWriterHandler.handle(ResponseWriterHandler.java:33)
	at org.jboss.resteasy.reactive.server.handlers.ResponseWriterHandler.handle(ResponseWriterHandler.java:15)
	at org.jboss.resteasy.reactive.common.core.AbstractResteasyReactiveContext.run(AbstractResteasyReactiveContext.java:141)
	at io.quarkus.vertx.core.runtime.VertxCoreRecorder$14.runWith(VertxCoreRecorder.java:548)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1478)
	at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)
	at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:29)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.lang.Thread.run(Thread.java:833)
	at com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:704)
	at com.oracle.svm.core.posix.thread.PosixPlatformThreads.pthreadStartRoutine(PosixPlatformThreads.java:202)
Caused by: java.lang.NoSuchMethodException: io.quarkus.registry.app.DatabaseRegistryClient.resolveCurrentPlatformsCatalog(java.lang.String)
	at java.lang.Class.getMethod(DynamicHub.java:2227)
	at org.jboss.resteasy.reactive.server.spi.ResteasyReactiveResourceInfo.getMethod(ResteasyReactiveResourceInfo.java:60)
	... 17 more
```


# How to reproduce: 

- Migrate to RestEasy Reactive - `git revert 428f579b1e86abe4181240d200504d850740844e`
- Build to native using `mvn clean package -Dnative`
- Set the following env vars: 
```
export TOKEN=test
export QUARKUS_DATASOURCE_USERNAME=admin
export QUARKUS_DATASOURCE_PASSWORD=mysecretpassword
export QUARKUS_DATASOURCE_JDBC_URL=jdbc:postgresql://localhost:5432/postgres
```

- Start a PostgreSQL DB (I'm using Docker on M1)
```
docker run -it --name some-postgres -p 5432:5432 -e POSTGRES_USER=admin -e POSTGRES_PASSWORD=mysecretpassword -e POSTGRES_DB=postgres arm64v8/postgres
```

- Start the application: 
```
target/quarkus-registry-app-999-SNAPSHOT-runner
```

- Populate the database running the script in https://github.com/quarkusio/quarkus-extension-catalog/blob/main/publish_local.sh

- Open http://localhost:8080/client/platforms

--- a/extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/main/java/io/quarkus/resteasy/reactive/server/deployment/ResteasyReactiveProcessor.java
+++ b/extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/main/java/io/quarkus/resteasy/reactive/server/deployment/ResteasyReactiveProcessor.java
@@ -670,6 +670,41 @@ public class ResteasyReactiveProcessor {
         serverSerializersProducer.produce(new ServerSerialisersBuildItem(serialisers));
     }
 
+    @BuildStep
+    public void additionalReflection(BeanArchiveIndexBuildItem beanArchiveIndexBuildItem,
+            SetupEndpointsResultBuildItem setupEndpointsResult, List<MessageBodyWriterBuildItem> messageBodyWriterBuildItems,
+            BuildProducer<ReflectiveClassBuildItem> producer) {
+        List<ResourceClass> resourceClasses = setupEndpointsResult.getResourceClasses();
+        IndexView index = beanArchiveIndexBuildItem.getIndex();
+
+        // when user provided MessageBodyWriter classes exist that do not extend ServerMessageBodyWriter, we need to enable reflection
+        // on every resource method, because these providers will be checked first and the JAX-RS API requires
+        // the method return type and annotations to be passed to the serializers
+        boolean serializersRequireResourceReflection = false;
+        for (var writer : messageBodyWriterBuildItems) {
+            if (writer.isBuiltin()) {
+                continue;
+            }
+            if ((writer.getRuntimeType() != null) && (writer.getRuntimeType() == RuntimeType.CLIENT)) {
+                continue;
+            }
+            ClassInfo writerClassInfo = index.getClassByName(DotName.createSimple(writer.getClassName()));
+            if (writerClassInfo == null) {
+                continue;
+            }
+            List<DotName> interfaceNames = writerClassInfo.interfaceNames();
+            if (!interfaceNames.contains(ResteasyReactiveServerDotNames.SERVER_MESSAGE_BODY_WRITER)) {
+                serializersRequireResourceReflection = true;
+                break;
+            }
+        }
+        if (serializersRequireResourceReflection) {
+            producer.produce(ReflectiveClassBuildItem
+                    .builder(resourceClasses.stream().map(ResourceClass::getClassName).toArray(String[]::new)).fields(false)
+                    .constructors(false).methods(true).build());
+        }
+    }
+
     @SuppressWarnings("unchecked")
     @BuildStep
     @Record(value = ExecutionTime.STATIC_INIT, useIdentityComparisonForParameters = false)

